Source/Core/InputCommon/ControlReference/ControlReference.cpp

```diff
@@ -48,6 +47,17 @@ ParseStatus ControlReference::GetParseStatus() const
   return m_parse_status;
 }
 
+std::string ControlReference::GetExpression() const
+{
+  return m_expression;
+}
+
+void ControlReference::SetExpression(std::string expr)
+{
+  m_expression = expr;
```

Comment:
What is your metric for 'worse' and how did you come to that conclusion (please don't say instruction count in the output, it alone isn't always a good one)? The move inlines more code than the copy because no resources need to be allocated. With a copy you have internal exception handling and calls to new and delete.